### ISSUE TITLE
SMAR-185 ⁃ Account for sensor distance from top of bin

### ISSUE THREAD

**Issue by thatdoogieguy** [thatdoogieguy/SST#691]
[~indr4_winata] we need to adjust the system full calcuations to take into account that the sensors might be placed some distance from the actual top of the bin.

I think the best method is we do this and build it into the container type model to accommodate for all variations in this distance the user could be faced with.

We would then need to base our calculations on 2 metrics.

# # Height of container from bottom to fill line.

1. Distance of sensor from top of fill line.

We would then be able to factor this into our measurements.

So basically if the sensor says rubbish is 50 CMs from the sensor, and the sensor distance is 20 CMs, then we know the rubbish 30 CMs from the top of the bin.

Let me know the impact of this change. It will be global. Don't change the raw data though.

┆Attachments: <a href="https://smartsensor.atlassian.net/secure/attachment/15200/Screen+Shot+2016-08-03+at+12.42.09+PM.png">Screen Shot 2016-08-03 at 12.42.09 PM.png</a>

**Comment by thatdoogieguy** [comment]
➤ User  commented:

[~yaroslav.zhurakovskiy]:
There is a change to some APIs.
They are:

  * Bin Types API and all API that have Bin Types with field depthWhenEmpty_cm information 
  * Allocate Sensor -> important one

Now to display the fill level, please use the following calculation:
Before
(depthWhenEmpty_cm - ultrasound) / depthWhenEmpty_cm
Now
(depthWhenEmpty_cm - ultrasound - distanceSensorToFillLine_cm) / depthWhenEmpty_cm

[~admin]: right now I put status "On Review". Maybe you want to add the same Jira for Yaroslav, or assign this to him and restart progress?

**Comment by thatdoogieguy** [comment]
➤ User  commented:

[~colby] can you please review changes in the Dashboard.
